app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventDetails/providers/InputFieldsProvider.kt

```diff
@@ -0,0 +1,360 @@
+    }
+
+    var expanded by remember { mutableStateOf(false) }
+
+    ExposedDropdownMenuBox(
+        expanded = expanded,
+        onExpandedChange = {},
+    ) {
+        InputDropDown(
+            modifier = modifier,
+            title = category.name,
+            state = if (detailsEnabled) {
+                InputShellState.UNFOCUSED
+            } else {
+                InputShellState.DISABLED
+            },
+            selectedItem = selectedItem,
+            onResetButtonClicked = {
+                onClearCatCombo(category)
+            },
+            onArrowDropDownButtonClicked = {
+                expanded = !expanded
+            },
+        )
+
+        if (expanded) {
+            if (category.optionsSize > DEFAULT_COUNT_LIMIT) {
+                onShowCategoryDialog(category)
+                expanded = false
+            } else {
+                DropdownMenu(
+                    modifier = modifier.exposedDropdownSize(),
+                    expanded = expanded,
+                    onDismissRequest = { expanded = false },
+                ) {
+                    val selectableOptions = category.options
+                        .filter { option ->
+                            option.access().data().write()
+                        }.filter { option ->
+                            option.inDateRange(currentDate)
+                        }.filter { option ->
+                            option.inOrgUnit(selectedOrgUnit)
+                        }
+                    selectableOptions.forEach { option ->
+                        val isSelected = option.displayName() == selectedItem
+                        DropdownMenuItem(
+                            modifier = Modifier.background(
+                                when {
+                                    isSelected -> SurfaceColor.PrimaryContainer
+                                    else -> Color.Transparent
+                                },
+                            ),
+                            content = {
+                                Text(
+                                    text = option.displayName() ?: option.code() ?: "",
+                                    color = when {
+                                        isSelected -> TextColor.OnPrimaryContainer
+                                        else -> TextColor.OnSurface
+                                    },
+                                )
+                            },
+                            onClick = {
+                                expanded = false
+                                selectedItem = option.displayName()
+                                onOptionSelected(option)
+                            },
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun ProvideCoordinates(
+    coordinates: EventCoordinates,
+    detailsEnabled: Boolean,
+    resources: ResourceManager,
+) {
+    InputCoordinate(
```

Comment:
Couldn't the event feature type be a polygon?

Reply:
let me check

Reply:
you were right, changed